Framework/PythonInterface/mantid/plots/axesfunctions.py

```diff
@@ -113,9 +113,16 @@ def _get_data_for_plot(axes, workspace, kwargs, with_dy=False, with_dx=False):
         workspace_index, distribution, kwargs = get_wksp_index_dist_and_label(
             workspace, axis, **kwargs)
         if axis == MantidAxType.BIN:
-            # Overwrite any user specified xlabel
-            axes.set_xlabel("Spectrum")
             x, y, dy, dx = get_bins(workspace, workspace_index, with_dy)
+            vertical_axis = workspace.getAxis(1)
+            if isinstance(vertical_axis, mantid.api.NumericAxis):
+                axes.set_xlabel(vertical_axis.getUnit().unitID())
+                values = vertical_axis.extractValues()
+                if isinstance(vertical_axis, mantid.api.BinEdgeAxis):
+                    # for bin edge axis we have one more edge than content
+                    values = (values[0:-1] + values[1:])/2.
+                x = values
```

Comment:
Should we not a include a check if it's a `SpectrumAxis` and set the x label to Spectrum?

Reply:
Never mind! I should have read the testing example and that test has already been covered!